### ISSUE TITLE
no-useless-escape should ignore line breaks in strings

### ISSUE THREAD

**Issue by AlfonsoML** [eslint/eslint#5689]
**What version of ESLint are you using?**
2.5.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
Use this:
`/*eslint no-useless-escape: 2 */`

**What did you do? Please include the actual source code causing the issue.**
Create a multiline string separated by `\` like this:

```js
/* exported selectorCss */
function selectorCss() {
	return '.someclass {\
		color: #000000;\
		opacity: 0.8;\
	}';
}
```
**What did you expect to happen?**
No errors or warnings

**What actually happened? Please include the actual, raw output from ESLint.**
```sh
  5:9  error  Unnecessary escape character: \
  no-useless-escape
  5:9  error  Unnecessary escape character: \
  no-useless-escape
  5:9  error  Unnecessary escape character: \
  no-useless-escape

✖ 3 problems (3 errors, 0 warnings)
```

**Comment by platinumazure** [comment]
Confirmed. This seems like a bug.

**Comment by michaelficarra** [comment]
https://github.com/eslint/eslint/pull/5632#issuecomment-201346520

**Comment by platinumazure** [comment]
Unfortunately that will come down to semantics. The rule is called "no-useless-escape"; escaping for a multiline string is unnecessary, but not useless-- the escape does have a real effect. But, of course, the message says unnecessary.

Maybe the rule could be enhanced to allow this particular case (and at least have a more useful error message). Changing to enhancement request.

**Comment by BYK** [comment]
I think having an `allowMultiline` or `ignoreMultiline` option would be good.

**Comment by mysticatea** [comment]
I noticed we have [no-multi-str] rule.
So I vote `no-useless-escape` allows multiline strings.
We can forbid multiline strings with [no-multi-str] rule.

[no-multi-str]: http://eslint.org/docs/rules/no-multi-str

**Comment by michaelficarra** [comment]
@mysticatea That works for me.

**Comment by mysticatea** [comment]
I think we can move forward this.

- This was intended, but we can change the behavior to allowing multiline strings as [no-multi-str] is covering this. I think this is not a breaking change since this is easing.

I will work on this within a few days.

[no-multi-str]: http://eslint.org/docs/rules/no-multi-str
